src/AzureAuth/CommandMain.cs

```diff
@@ -234,6 +234,20 @@ public static string PrefixedPromptHint(string promptHint)
             return $"{PromptHintPrefix}: {promptHint}";
         }
 
+        /// <summary>
+        /// Get the correct CombinedAuthMode depending on env variables to disable interactive auth modes.
+        /// </summary>
+        /// <returns>AuthModes.</returns>
+        public AuthMode GetCombinedAuthMode()
+        {
+            if (this.InteractiveAuthDisabled())
+            {
+                return AuthMode.IWA;

```

Comment:
Referencing the actual `AuthMode.IWA` enum variant is only possible on Windows, since the enum is conditionally compiled. The IsEnabled method exists on both platforms is hardcoded to return false on non-Windows platforms.

Reply:
I like managing the modes in this method, but it will have to be platform specific. You could return a specific mode if windows, and no mode (0 as this enum) if Mac.

Reply:
Updated to return platform specific auth mode